akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpRequestRendererFactory.scala

```diff
@@ -99,7 +99,7 @@ private[http] class HttpRequestRendererFactory(userAgentHeader: Option[headers.`
       }
 
     def renderContentLength(contentLength: Long) =
-      if (method.isEntityAccepted) r ~~ `Content-Length` ~~ contentLength ~~ CrLf else r
+      if (method.isEntityAccepted && (contentLength > 0 || !method.isSafe)) r ~~ `Content-Length` ~~ contentLength ~~ CrLf else r
```

Comment:
good point. maybe we should change `isEntityAccepted` from Boolean to "yes/no/unkilely" (GET/DELETE = unilkely) and omit rendering of `Content-Length` in the unlikely case if it's zero.